bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/crypto/EncryptionHandler.java

```diff
@@ -0,0 +1,307 @@
+        // fill array to 32th byte with 0xEE
+        byte[] filledArray = fillArray(input);
+
+        int[] temp = { 0, 0, 0, 0 };
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 4; j++) {
+                expandedKey[j * 4 + i] = filledArray[i * 4 + j] & 0xFF;
+            }
+            for (int j = 0; j < 4; j++) {
+                expandedKey[j * 4 + i + 16] = filledArray[i * 4 + j + 16] & 0xFF;
+            }
+        }
+
+        final int[] s = EncryptionHandlerConstants.S;
+        for (int i = 8; i < 60; i++) {
+            for (int j = 0; j < 4; j++) {
+                temp[j] = expandedKey[(((i - 1) & 0xfc) << 2) + ((i - 1) & 0x03) + j * 4];
+            }
+
+            if (i % 8 == 0) {
+                for (int j = 0; j < 4; j++) {
+                    temp[j] = s[temp[j]];
+                }
+                int tmp = temp[0];
+
+                for (int j = 1; j < 4; j++) {
+                    temp[j - 1] = temp[j];
+                }
+
+                temp[3] = tmp;
+                temp[0] ^= EncryptionHandlerConstants.RCON[(i / 8 - 1)];
+            } else if (i % 8 == 4) {
+                for (int j = 0; j < 4; j++) {
+                    temp[j] = s[temp[j]];
+                }
+            }
+
+            for (int j = 0; j < 4; j++) {
+                expandedKey[((i & 0xfc) << 2) + (i & 0x03)
+                        + j * 4] = expandedKey[(((i - 8) & 0xfc) << 2) + ((i - 8) & 0x03) + j * 4] ^ temp[j];
+            }
+        }
+    }
+
+    private int gmul(int c, int b) {
+        int s;
+        int q;
+        int z = 0;
+        s = lTable[c] + lTable[b];
+        s %= 255;
+        /* Get the antilog */
+        s = aTable[s];
+        /*
+         * Now, we have some fancy code that returns 0 if either
+         * a or b are zero; we write the code this way so that the
+         * code will (hopefully) run at a constant speed in order to
+         * minimize the risk of timing attacks
+         */
+        q = s;
+        if (c == 0) {
+            s = z;
+        } else {
+            s = q;
+        }
+        if (b == 0) {
+            s = z;
+        } else {
+            q = z;
+        }
+        return s;
+    }
+
+    private void generateTables() {
+        int a = 1;
+        int d;
+        for (int index = 0; index < 255; index++) {
+            aTable[index] = a & 0xFF;
+            /* Multiply by three */
+            d = (a & 0x80) & 0xFF;
+            a <<= 1;
+            if (d == 0x80) {
+                a ^= 0x1b;
+                a &= 0xFF;
+            }
+            a ^= aTable[index];
+            a &= 0xFF;
+            /* Set the log table value */
+            lTable[aTable[index]] = index & 0xFF;
+        }
+        aTable[255] = aTable[0];
+        lTable[0] = 0;
+    }
+
+    private void sBox(int[] a, int[] box) {
+        for (int i = 0; i < 16; i++) {
+            a[i] = box[a[i]];
+        }
+    }
+
+    private void mixColumn(int[] a) {
```

Comment:
```suggestion
    private static void mixColumn(int[] a) {
```

Reply:
In general this comment applies to all changes of methods in handler to static. What is the benefit of making static use of methods? I would rather prefer to encapsulate the key inside a singleton instance of the class and call these methods in non-static way. Maybe it relates ot the use of singleton pattern... Probably we can discuss that in the other thread related to the usage of singleton.

Reply:
The main benefits are architectural, the performance improvements are very minimal.